io-engine/src/grpc/v1/replica.rs

```diff
@@ -555,6 +555,26 @@ impl ReplicaRpc for ReplicaService {
                             })
                         }
                     };
+                    // validate snapshot name
+                    if let Some(_r) = UntypedBdev::lookup_by_name(&args.snapshot_name) {
+                        return Err(LvsError::Invalid {
+                            source: Errno::EEXIST,
```

Comment:
Please map these source errors to grpc status code on the From<Lvs> to tonic::Status, I believe existing ones are already missing, and get converted to the wrong grpc status code.

Reply:
> Please map these source errors to grpc status code on the From to tonic::Status, I believe existing ones are already missing, and get converted to the wrong grpc status code.

Updated Code with new gRPC error code.